packages/server/src/middleware/migrations.ts

```diff
@@ -0,0 +1,143 @@
+import { context, locks, queue } from "@budibase/backend-core"
+import { UserCtx, Document, LockName, LockType } from "@budibase/types"
+import Bull, { Job } from "bull"
+
+export const MIGRATIONS: Record<string, { migration: () => Promise<void> }> = {
+  "20231122115100": {
+    migration: async () => {
+      await new Promise<void>(r => {
+        setTimeout(() => {
+          console.error("a")
+          r()
+        }, 10000)
+      })
+    },
+  },
+  "20231122115532": {
+    migration: async () => {
+      await new Promise<void>(r => {
+        setTimeout(() => {
+          console.error("b")
+          r()
+        }, 10000)
+      })
+    },
+  },
+}
+
+export interface MigrationDoc extends Document {
+  version: string
+}
+
+const appQueues: Record<string, Bull.Queue<any>> = {}
+
+async function processMessage(job: Job) {
```

Comment:
I mentioned it on the design doc, but it would be nice to be able to store migration logs in a way that's easier for us or the user to fetch them to try and debug what went wrong if a migration fails. Not sure how easy that would be today, or if we would want to put something special in place to capture and store them.